### PR TITLE
Add Linux support for Bats tests

### PR DESCRIPTION
**Pull request by thehejik**
Here is my initial attempt to make the new bats tests running on x86_64 Linux

All the tests are Passing

### CODE REVIEW

**Comment by thehejik** [comment]
@jandubois please take a look & comment

File: bats/tests/helpers/vm.bash [hunk 1]
```diff
@@ -9,19 +9,22 @@ assert_rd_is_stopped() {
...
+    is_linux && ! pgrep rancher-desktop &>/dev/null
 }
 
 wait_for_shutdown() {
     try --max 18 --delay 5 assert_rd_is_stopped
 }
 
 factory_reset() {
-    run $RDCTL shutdown
+    $RDCTL shutdown
```
Comment (jandubois) [inline bats/tests/helpers/vm.bash:21]: Why did you remove the call to `run`. I put it there so the function doesn't fail if RD already isn't running anymore.

Comment (thehejik) [reply bats/tests/helpers/vm.bash]: You are right, reverting back.

File: bats/tests/containers/platform.bats [hunk 2]
```diff
@@ -30,6 +30,7 @@ check_uname() {
 }
 
 @test 'deploy arm64 container' {
+    is_linux && skip
```
Comment (jandubois) [inline bats/tests/containers/platform.bats:33]: Why is this necessary? This is supposed to work everywhere because we configure qemu-static emulation for the arm64 on the x86_64 VM (I think we are missing this in the WSL distro, but Linux uses the same ISO as macOS). Did this actually fail for you?

Comment (thehejik) [reply bats/tests/containers/platform.bats]: Actually it works, I'll remove that line.

**Comment by thehejik** [comment]
> You will have to rebase your PR branch accordingly to get rid of the merge conflicts...

Done.

**Review by jandubois — APPROVED** [review]
Thanks!
